gdc_client/upload/client.py

```diff
@@ -564,8 +553,17 @@ def upload_parts(self):
 
             for future in as_completed(future_to_part_number):
                 part_number = future_to_part_number[future]
-                log.debug("Part: {} is done".format(part_number))
-                pbar.update()
+                """
+                    upload_multipart() returns True on success or False on failure
+                    upload_multipart() is responsible for catching exceptions,
+                    so no exception should be re-raised here
+                """
+                if future.result():
+                    log.debug("Part: {} is done".format(part_number))
+                    self.completed += 1
```

Comment:
:100: I also prefer updating `completed` in this manner, its more intuitive and probably better in terms of thread safety.